lib/services/role.go

```diff
@@ -1772,15 +1772,15 @@ func (set RoleSet) CertificateExtensions() []*types.CertExtension {
 
 // SessionRecordingMode returns the recording mode for a specific service.
 func (set RoleSet) SessionRecordingMode(service constants.SessionRecordingService) constants.SessionRecordingMode {
-	defaultValue := constants.SessionRecordingModeBestEffort
-	useDefault := true
+	defaultValue := constants.SessionRecordingModeUnspecified
+	serviceValue := constants.SessionRecordingModeUnspecified
 
 	for _, role := range set {
 		recordSession := role.GetOptions().RecordSession
 
-		// If one of the default values is "strict", set it as the value.
-		if recordSession.Default == constants.SessionRecordingModeStrict {
-			defaultValue = constants.SessionRecordingModeStrict
+		// If one of the default values is stricter, set it as the value.
+		if isSessionRecordingModeStricter(defaultValue, recordSession.Default) {
+			defaultValue = recordSession.Default
```

Comment:
super nit: probably can combine these two lines in one like
```suggestion
		defaultValue = pickStricterSessionRecordingMode(defaultValue,  recordSession.Default)
```

Reply:
Could we also initialize the `roleMode` to unspecified?

Reply:
I agree that it reduces repeating, but I think the `isSessionRecordingModeStricter` is easier to understand and more straightforward than a function that selects them.